src/http/http-request.js

```diff
@@ -0,0 +1,226 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import UploadBase from '../upload-base';
+import { DefaultValues } from '../constants';
+import UploadError from '../upload-error';
+import ErrorCodes from '../error-codes';
+import { Readable } from 'stream';
+
+/**
+ * Represents a request that can be sent via HTTP using the HttpClient. Options on
+ * the request can be set via chaining its "with" methods together.
+ */
+class HttpRequest extends UploadBase {
+
+    /**
+     * Constructs a new request using the given values. Default method
+     * is GET.
+     * @param {object} options 
+     * @param {string} url URL to which the request will be sent.
+     */
+    constructor(options, url) {
+        super(options);
+
+        this.requestOptions = {
+            url,
+            method: HttpRequest.Method.GET,
+            timeout: DefaultValues.REQUEST_TIMEOUT,
+        };
+        this.headers = {};
+        this.cancelId = '';
```

Comment:
Why an empty `cancelId` rather than `null`?

Reply:
I think it's just a matter of preference. It seems pretty common in javascript to test variables with `if (cancelId)`, which would resolve to the same thing regardless of whether it's `''` or `null`.